analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -367,30 +367,29 @@ open class Npm(
 
             val project = parseProject(definitionFile)
 
-            // Optional dependencies are just like regular dependencies except that NPM ignores failures when
-            // installing them (see https://docs.npmjs.com/files/package.json#optionaldependencies), i.e. they are
-            // not a separate scope in our semantics.
-            buildDependencyGraphForScopes(
-                project,
-                workingDir,
-                setOf(DEPENDENCIES_SCOPE, OPTIONAL_DEPENDENCIES_SCOPE),
-                DEPENDENCIES_SCOPE
-            )
+            val scopeNames = listOfNotNull(
```

Comment:
Nit: Why not straight construct a `sortedSetOf()` to avoid the conversion below?

Reply:
Ah, sorry, scratch that. We have no "NotNull" version of that.